tests/test_s3/test_s3.py

```diff
@@ -133,6 +136,24 @@ def test_my_model_save():
     assert body == "is awesome"
 
 
+@mock_s3
+def test_object_metadata():
+    """Metadata keys can contain certain special characters like underscore and dot"""
+    # Create Bucket so that test can run
+    conn = boto3.resource("s3", region_name=DEFAULT_REGION_NAME)
+    conn.create_bucket(Bucket="mybucket")
+    ####################################
+
+    metadata = {"meta": "simple", "my_meta": "underscore", "meta.data": "namespaced"}
```

Comment:
The tests are currently failing because of the underscore - would you mind changing this to `"my-meta": "dash"`? Happy to merge once the build passes

(Some background on why/how this is  failing can be found here: https://github.com/spulec/moto/issues/2054)

Reply:
Got it, makes sense. test updated.